Fix PyPI build directory in release workflow

## Description

The PyPI publishing job in the release workflow is failing because maturin is being run from the wrong directory.

## Current Error

```
Run maturin build --release --features python
💥 maturin failed
  Caused by: Can't find /home/runner/work/jcl/jcl/bindings/python/Cargo.toml (in /home/runner/work/jcl/jcl/bindings/python)
Error: Process completed with exit code 1.
```

## Root Cause

The workflow currently has:
```yaml
- name: Build Python wheels
  working-directory: bindings/python
  run: maturin build --release --features python
```

Maturin needs to run from the **root directory** where the main `Cargo.toml` is located, not from `bindings/python/`. The `pyproject.toml` in `bindings/python/` references the `python` feature from the root Cargo.toml.

## Proposed Fix

Remove the `working-directory` from the maturin build step and specify the manifest path:

```yaml
- name: Build Python wheels
  run: maturin build --release --features python --manifest-path Cargo.toml
```

The wheels will still be output to the correct location that the publish step expects.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -202,7 +202,7 @@ jobs:
 
       - name: Build Python wheels
         working-directory: bindings/python
-        run: maturin build --release --features python
+        run: maturin build --release
 
       - name: Publish to PyPI (with trusted publishing)
         uses: pypa/gh-action-pypi-publish@release/v1

--- a/bindings/python/Cargo.toml
+++ b/bindings/python/Cargo.toml
@@ -1,0 +1,17 @@
+[package]
+name = "jcl-python"
+version = "1.0.0"
+edition = "2021"
+authors = ["Hemmer IO <info@hemmer.io>"]
+description = "Python bindings for JCL (Jack-of-All Configuration Language)"
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/hemmer-io/jcl"
+
+[lib]
+name = "jcl"
+crate-type = ["cdylib"]
+
+[dependencies]
+jcl = { path = "../..", version = "1.0.0", features = ["python"] }
+pyo3 = { version = "0.20", features = ["extension-module"] }
+serde_json = "1.0"

